internal/services/keyvault/client/helpers.go

```diff
@@ -111,6 +113,32 @@ func (c *Client) Exists(ctx context.Context, keyVaultId commonids.KeyVaultId) (b
 	return true, nil
 }
 
+func (c *Client) KeyVaultIDFromBaseURLByGraph(ctx context.Context, client *graph.ResourcesClient, name string) (string, interface{}) {
+	if name == "" {
+		return "", nil
+	}
+	req := graph.QueryRequest{
+		Query: fmt.Sprintf("where type =~ 'Microsoft.KeyVault/vaults' and name =~ '%s'", name),
+	}
+	resp, err := client.Resources(ctx, req)
+	if err != nil {
+		return "", err
+	}
+
+	if resp.Model != nil && resp.Model.Data != nil {
+		if list, ok := resp.Model.Data.([]interface{}); ok && len(list) == 1 {
```

Comment:
What if it returns `0` or `2`? Is returning the `no such keyvault in resource graph by query` the correct behavior here?

Reply:
There shoud be `1` or `0` resource for keyvault resource of the same name.